kornia/geometry/camera/pinhole.py

```diff
@@ -61,6 +62,21 @@ def _check_valid_shape(data: torch.Tensor, data_name: str) -> bool:
             )
         return True
 
+    @staticmethod
+    def _check_consistent_device(data_iter: Iterable[torch.Tensor]) -> bool:
+        first = data_iter[0]
```

Comment:
Will have a look

Reply:
I think this error could be because I was using typing `Iterable` for `data_iter`, but sent over a list to the check consistency function. I changed the typing of this argument to `List`. However, since I did not get any `mypy` errors with the previous version, I'm not sure that this is what caused the issue.